test/alternator/test_batch.py

```diff
@@ -375,6 +378,43 @@ def test_batch_write_item_large(test_table_sn):
     assert full_query(test_table_sn, KeyConditionExpression='p=:p', ExpressionAttributeValues={':p': p}
         ) == [{'p': p, 'c': i, 'content': long_content} for i in range(25)]
 
+# Test if client breaking connection during HTTP response
+# streaming doesn't break the server.
+def test_batch_write_item_large_broken_connection(test_table_sn, dynamodb_function_scoped):
+    fn_name = sys._getframe().f_code.co_name
+    # This disruption doesn't always work so we repeat it.
+    for _ in range(1, 20):
+        p = random_string()
+        long_content = random_string(100)*500
+        write_reply = test_table_sn.meta.client.batch_write_item(RequestItems = {
+            test_table_sn.name: [{'PutRequest': {'Item': {'p': p, 'c': i, 'content': long_content}}} for i in range(25)],
+        })
+        assert 'UnprocessedItems' in write_reply and write_reply['UnprocessedItems'] == dict()
+
+        read_fun = urllib3.HTTPResponse.read_chunked
+        triggered = False
+        def broken_read_fun(self, amt=None, decode_content=None):
+            ret =  read_fun(self, amt, decode_content)
+            st = traceback.extract_stack()
+            #print(st, fn_name)
+            # Try to not disturb other tests if executed in parallel
+            if fn_name in str(st):
+                self._fp.fp.raw.close() # close the socket
+                nonlocal triggered
+                triggered = True
+            return ret
+        urllib3.HTTPResponse.read_chunked = broken_read_fun
+
+        try:
+            # Our monkey patched read_chunked function will make client unusable
```

Comment:
You said that you used a separate dynamodb "resource" because of this monkey-patching. But I don't understand how the separate resource helps... You're monkey-patching a global urllib3 function, which all boto3 code supposedly uses, in all sessions.
Maybe you meant that you need the new session because of the `close()` call **inside** the monkey-patched function, not because of the monkey-patching?

Reply:
yes, as wrote "monkey patched read_chunked function will make client unusable", the function `read_chunked` breaks the session (due to close but that's detail)